### ISSUE TITLE
[Question] How to improve performance of the model (log loss)?

### ISSUE THREAD

**Issue by kuroahna** [open-spaced-repetition/fsrs4anki#190]
**Research**
- [x] I have checked the [FAQ](https://github.com/open-spaced-repetition/fsrs4anki/wiki/FAQ) and could not find an answer to my question
- [x] I have read the [wiki](https://github.com/open-spaced-repetition/fsrs4anki/wiki) and still felt confused
- [x] I have searched for similar existing questions here

**Question**

If I understand correctly, in the optimizer, we train and evaluate the model at the end to see how well it performs against our revlog.

In the optimizer, I am getting
```
100%
319978/319978 [13:56<00:00, 1049.71it/s]
Loss before training: 0.4519

100%
319978/319978 [11:11<00:00, 1951.36it/s]
Loss after training: 0.4105
```
![image](https://user-images.githubusercontent.com/85209455/226711792-cf3ef768-457f-4295-9160-5e67622317d1.png)

So before training it has a log loss of 0.4519, and after training it has a log loss of 0.4105. I'm not too familiar with what log loss is, but according to https://towardsdatascience.com/intuition-behind-log-loss-score-4e0c9979680a , they say

>Log-loss is indicative of how close the prediction probability is to the corresponding actual/true value (0 or 1 in case of binary classification). **The more the predicted probability diverges from the actual value, the higher is the log-loss value.**

So lower log loss values means that the model is performing well.

Is there a way to get better performance? Does it need more iterations? Larger/smaller learning rate (default is `5e-4`)?

My parameters are 

```
[1.2958, 0.9669, 5.428, -1.248, -1.0036, 0.0257, 1.2816, -0.0125, 0.7222, 1.7748, -0.4153, 0.6615, 0.7742]
```

My revlog is fairly large, I have learned 25,000 cards and it says
```
100%
319978/319978 [00:56<00:00, 15882.08it/s]

Tensorized!

pre-train: 100%
24915/24915 [00:22<00:00, 1222.99it/s]

w: [1.2973, 0.9661, 5.0, -0.5, -0.5, 0.2, 1.4, -0.12, 0.8, 2.0, -0.2, 0.2, 1.0]

train: 100%
295063/295063 [31:51<00:00, 175.43it/s]

iteration: 1
w: [1.2972, 0.9662, 4.9984, -0.4984, -0.4984, 0.2016, 1.4016, -0.1184, 0.8016, 2.0016, -0.1984, 0.2016, 1.0016]
iteration: 29507
w: [1.2958, 0.9669, 5.3661, -1.1155, -0.7993, 0.0307, 1.2186, -0.0149, 0.6264, 1.9242, -0.2801, 0.4937, 1.0011]
iteration: 59013
w: [1.2958, 0.9669, 5.4537, -1.2492, -0.9465, 0.0142, 1.2804, -0.0137, 0.6951, 1.8297, -0.3733, 0.5586, 0.9535]
iteration: 88519
w: [1.2958, 0.9669, 5.4437, -1.2469, -1.0196, 0.0082, 1.314, -0.0151, 0.733, 1.7821, -0.4214, 0.5857, 0.9113]
iteration: 118025
w: [1.2958, 0.9669, 5.4529, -1.2656, -1.0098, 0.0317, 1.2845, -0.01, 0.7078, 1.811, -0.3916, 0.6799, 0.8939]
iteration: 147531
w: [1.2958, 0.9669, 5.469, -1.3217, -0.9909, 0.0102, 1.2798, -0.0172, 0.7056, 1.7917, -0.4097, 0.6705, 0.8465]
iteration: 177037
w: [1.2958, 0.9669, 5.4721, -1.2896, -0.9882, 0.0066, 1.2844, -0.01, 0.7166, 1.7813, -0.4161, 0.6214, 0.8602]
iteration: 206543
w: [1.2958, 0.9669, 5.4389, -1.2586, -0.9798, 0.0107, 1.2991, -0.0102, 0.7358, 1.7884, -0.4068, 0.6385, 0.8364]
iteration: 236049
w: [1.2958, 0.9669, 5.4359, -1.2398, -1.0338, 0.0174, 1.2849, -0.0103, 0.7249, 1.7748, -0.4183, 0.6208, 0.817]
iteration: 265555
w: [1.2958, 0.9669, 5.4199, -1.2309, -1.0045, 0.0071, 1.2895, -0.0144, 0.7301, 1.7955, -0.398, 0.6458, 0.7933]
iteration: 295061
w: [1.2958, 0.9669, 5.4281, -1.2479, -1.0037, 0.0255, 1.2815, -0.0126, 0.7221, 1.7746, -0.4155, 0.6615, 0.774]

Training finished!
```
when training, so there should be enough training data

**Comment by L-M-Sherlock** [comment]
By the way, there are many tricks in deep learning. Here is a cheat sheet: https://stanford.edu/~shervine/teaching/cs-230/cheatsheet-deep-learning-tips-and-tricks
